src/utils/power_flow/make_pf.jl

```diff
@@ -0,0 +1,142 @@
+"""
+    make_pf(sys)
+
+Create the objects needed to solve an powerflow case using NLsolve.jl solvers. Returns 
+an anonymous function with the powerflow equations, initial conditions and a dict to link the
+solutions to the original system. Only supports systems with a single generator per bus and 
+currently doesn't support distributes slack buses and doesn't enforce reactive power limits. 
+
+## Example
+```julia
+pf!, x0, res_ref = make_pf(sys)  
+res = NLsolve.nlsolve(pf!, x0)
+```
+
+# Arguments
+    * `sys`::System : a PowerSystems.jl system
+
+"""
+
+macro fn(expr::Expr)
+    @assert expr.head in (:function, :->)
+    name = gensym()
+    args = expr.args[1]
+    args = typeof(args) == Symbol ? [args] : args.args
+    body = expr.args[2]
+    @eval $name($(args...)) = $body
+    name
+  end
+
+function make_pf(system)
+    buses = sort(collect(get_components(Bus, system)), by = x -> x.number)
+    bus_count = length(buses)
+    #assumes the ordering in YBus is the same as in the buses.
+    Yb = Ybus(system)
+
+    #internally allocate matrices
+    internal = quote
+        V = Vector{ComplexF64}(undef, $bus_count)
+        Vc = Vector{ComplexF64}(undef, $bus_count)
+        P_bal = Vector{Float64}(undef, $bus_count)
+        Q_bal = Vector{Float64}(undef, $bus_count)		
+    end
+
+    var_count = 1
+	res_dict = Dict{String, Vector{Tuple{Symbol, Int}}}() 
+	x0 = Array{Float64}(undef, bus_count*2)	
+    
+	for (ix, b) in enumerate(buses)
+         #Gets relevant data about the system, changes with the system that goes in.
+		 generators = [g for g in get_components(Generator, system) if g.bus == b]
```

Comment:
There is no need to create an array. Just iterate until you either find a match or get to the end.

Reply:
ok, I will update this